docs/concepts/simulation.md

```diff
@@ -0,0 +1,72 @@
+- `AppStateDeterminism`: Checks that all the nodes return the same values, in the same order.
+- `AppImportExport`: The simulator exportes the initial app state and then it creates a new app with the exported `genesis.json` as an input, checking for inconsistencies between the stores.
+- `AppSimulationAfterImport`: Queues two simulations together. The first one provides the app state (_i.e_ genesis) to the second. Useful to test software upgrades or hard-forks from a live chain.
+- `BenchmarkInvariants`: An invariant checks for differences between the values that are on the store and the passive tracker. Eg: total coins held by accounts vs total supply tracker.
+
+Each simulation must receive a set of inputs (_i.e_ flags) such as the number of blocks that the simulation is run, seed, block size, etc.
+Check the full list of flags [here](https://github.com/cosmos/cosmos-sdk/blob/adf6ddd4a807c8363e33083a3281f6a5e112ab89/simapp/sim_test.go#L34-L50).
+
+## Simulator Modes
+
+In addition to the various inputs and commands. The simulator runs in three modes:
+
+1. Completely random where the initial state, module parameters and simulation parameters are **pseudo-randomly generated**.
+2. From a `genesis.json` file where the initial state and the module parameters are defined.
+This mode is helpful for running simulations on a known state such as a live network export where a new (mostly likely breaking) version of the application needs to be tested.
+3. From a `params.json` file where the initial state is pseudo-randomly generated but the module and simulation parameters can be provided manually.
+This allows for a more controlled and deterministic simulation setup while allowing the state space to still be pseudo-randomly simulated. The list of available parameters is listed [here](https://github.com/cosmos/cosmos-sdk/blob/adf6ddd4a807c8363e33083a3281f6a5e112ab89/x/simulation/params.go#L170-L178).
+
+::: tip
+These modes are not excluding between each other. So you can for example run a randomly generated genesis state (`1`) with manually generated simulation params (`3`).
+:::
+
+## Usage
+
+This is a general example of how simulations are run. For more specific examples check the SDK [Makefile](https://github.com/cosmos/cosmos-sdk/blob/adf6ddd4a807c8363e33083a3281f6a5e112ab89/Makefile#L88-L123).
+
+```bash
+ $ go test -mod=readonly github.com/cosmos/cosmos-sdk/simapp \
+  -run=TestApp<simulation_command> \
+  ...<flags>
+  -v -timeout 24h
+```
+
+## Debugging Tips
+
+<!-- TODO: add output image -->
+
+Here are some suggestions when encountering a simulation failure:
+
+- Export the app state at the height were the failure was found. You can do this by passing the `-ExportStatePath` flag to the simulator.
+- Use `-Verbose` logs. They could give you a better hint on all the operations involved.
+
+- Reduce the simulation `-Period`. This will run the invariants checks more frecuently.
+- Print all the invariants at once with `-PrintAllInvariants`.
```

Comment:
```suggestion
- If an invariant breaks, you can print all the invariants at once with `-PrintAllInvariants`. It will print both passing and failing invariants.
```

Reply:
> It will print both passing and failing invariants. 

Uh? Doesn't it only print the failed invariants?